index.js

```diff
@@ -45,48 +54,67 @@ function buildMemoizer (options) {
   const promiseCache = {}
 
   // check args
-  if (typeof options !== 'object') {
+  if (!options || (options && typeof options !== 'object')) {
     throw new Error('options of type object expected')
   }
   if (typeof options.cachePath !== 'string') {
     throw new Error('option cachePath of type string expected')
   }
+  options = { ...serializer, ...options }
 
   // check for existing cache folder, if not found, create folder, then resolve
   function initCache (cachePath) {
     return new Promise(function (resolve, reject) {
-      return mkdirp(cachePath).then(() => {
-        resolve()
-      }).catch(reject)
+      return fs.ensureDir(cachePath, { recursive: true })
+        .then(() => {
+          resolve()
+        })
+        .catch((err) => {
+          // ! NOTE we should handle this case, intentionally;
```

Comment:
@borisdiakur thoughts?

Reply:
I’m not sure if I get it right. I see three possible cases here:

1. The path points to an existing directory -> resolves in the then-block
2. The path points to a file -> rejects in the if-block
2. The path does not exist yet -> also resolves in the then-block

Reply:
True. I'm not sure how exactly the `ensureDir` works, I should check. But for some reason it throws when directory exist (EEXIST) which I don't see why we should care and can make it `resolve()` in that case.

Reply:
I think the catch block should look like this:

```js
.catch((err) => {
  if (err && err.code === 'EEXIST') {
    reject(err)
    return
  }
  reject(err)
})
```

Reply:
@borisdiakur, nope. It should be `if err -> resolve()`. That's the whole point of this catch and comment in the first place. We should not care, we just need the directories created, there's no problem for us if it already exists.

Reply:
Ok, feels weird, but I get it. Just worried, because I don’t know if there might be other error codes that we should not ignore. For instance: what about ENOTDIR? See https://www-numi.fnal.gov/offline_software/srt_public_context/WebDocs/Errors/unix_system_errors.html
I haven’t done any research / testing or anything, it’s just unpleasant gut feeling.